File: Polygon.py
import math

class Polygon:
    name = ''
    sides = 0
    sideLength = 0

    def __init__(self, name, sides, sideLength):
        self.name = name
        self.sides = sides
        self.sideLength = sideLength

    def getName(self):
        pass

    def getPerimeter(self):
        return self.sideLength * self.sides

    def getArea(self):
        return 'Unknown'

    def getAngle(self):
        return (self.sides - 2) * 180 / self.sides

    def __str__(self):
        return self.name + " (" + str(self.sideLength) + ")"

class Triangle(Polygon):
    def __init__(self, sideLength):
        super().__init__('Triangle', 3, sideLength)

    def getArea(self):
        base = self.sideLength;
        height = (self.sideLength / 2) * math.sqrt(3);
        area = base * height / 2
        return area

class Square(Polygon):
    def __init__(self, sideLength):
        super().__init__('Square', 4, sideLength)

    def getArea(self):
        side = self.sideLength;
        area = side * side
        return area

class Pentagon(Polygon):
    def __init__(self, sideLength):
        super().__init__('Pentagon', 5, sideLength)

class Octagon(Polygon):
    def __init__(self, sideLength):
        super().__init__('Octagon', 8, sideLength)

class Nonagon(Polygon):
    def __init__(self, sideLength):
        super().__init__('Nonagon', 10, sideLength)

triangle = Triangle(10)
square = Square(10)
pentagon = Pentagon(10)
octagon = Octagon(10)
nonagon = Nonagon(10)
for shape in [ triangle, square, pentagon, octagon, nonagon ]:
    print(shape)
    perimeter = shape.getPerimeter()
    angle = shape.getAngle()
    area = shape.getArea()
    print('perimeter: ' + str(perimeter))
    print('angle: ' + str(angle))
    print('area: ' + str(area))
    print()

File: Polygon.java
import java.lang.Math;


public abstract class Polygon {
    protected String name = null;
    protected int sides = 0;
    protected int sideLength = 0;

    public Polygon(String name, int sides, int sideLength) {
        this.name = name;
        this.sides = sides;
        this.sideLength = sideLength;
    }

    public String getName() {
        return name;
    }

    public int getSides() {
        return sides;
    }

    public int getSideLength() {
        return sideLength;
    }

    public int getPerimeter() {
        return sideLength * sides;
    }

    public int getAngle() {
        return (sides - 2) * 180 / sides;
    }

    public abstract int getArea();

    public String toString() {
        return name + "(" + sideLength + ")";
    }

    public static void main(String[] args) {
        Triangle triangle = new Triangle(5);
        System.out.println(triangle);
        System.out.println("getName(): " + triangle.getName());
        System.out.println("getSides(): " + triangle.getSides());
        System.out.println("getSideLength(): " + triangle.getSideLength());
        System.out.println("getArea(): " + triangle.getArea());
        System.out.println("getPerimeter(): " + triangle.getPerimeter());
        System.out.println("getAngle(): " + triangle.getAngle());
    }
}


class Triangle extends Polygon {
    public Triangle(int sideLength) {
        super("Triangle", 3, sideLength);
    }

    public int getArea() {
        int base = sideLength;
        int height = (sideLength / 2) * (int) Math.sqrt(3);
        int area = base * height / 2;
        return area;
    }

}
